src/components/Login/Login.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import style from './style.module.css';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useHistory } from 'react-router-dom';
+import { useAuth } from '../Contexts/AuthContext';
+
+function Login(props) {
+    const [error, setError] = useState('');
+    const {currentUser, login} = useAuth();
+    const [loading, setLoading] = useState(false);
+    const history = useHistory();
+
+
+    const schema = yup.object().shape({
+        email: yup.string().max(20).required(),
+        password: yup.string().max(15).required(),
+    })
+
+    const {register, handleSubmit, formState: { errors }} = useForm({resolver: yupResolver(schema)});
+
+
+    const onSubmit = async (data) => {
+
+        try {
+            setError('');
+            setLoading(true);
+            await login(data.email, data.password);
+            history.push("/dashboard");
+            console.log(currentUser);
+        } catch {
+            setError('Failed to create an account');
+            console.log(data.email, data.password, currentUser)
+        }
+        setLoading(false);
+    }
+    console.log(errors)
+    
+
+    return (
+        <div>
+            <h1>Login</h1>
+            <form className={style.form} onSubmit={handleSubmit(onSubmit)}>
+
+                <div>
+                    <input type="text" placeholder="Email" {...register('email', {pattern: {value: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/, message: 'invalid email'},  required: true})} name="email"/>
```

Comment:
Don't add the email pattern here. You can set a const with it and set it up in Yup resolver itself.